core-image.html

```diff
@@ -195,49 +310,65 @@
       this.style.height = isNaN(this.height) ? this.height : this.height + 'px';
     },
 
-    update: function() {
-      this.style.backgroundSize = this.sizing;
-      this.style.backgroundPosition = this.sizing ? this.position : null;
-      this.style.backgroundRepeat = this.sizing ? 'no-repeat' : null;
-      if (this.preload) {
-        if (this.fade) {
-          if (!this._placeholderEl) {
-            this._placeholderEl = this.shadowRoot.querySelector('#placeholder');
-          }
-          this._placeholderEl.style.backgroundSize = this.sizing;
-          this._placeholderEl.style.backgroundPosition = this.sizing ? this.position : null;
-          this._placeholderEl.style.backgroundRepeat = this.sizing ? 'no-repeat' : null;
-          this._placeholderEl.classList.remove('fadein');
-          this._placeholderEl.style.backgroundImage = (this.load && this.placeholder) ? 'url(' + this.placeholder + ')': null;
-        } else {
-          this._setSrc(this.placeholder);
-        }
-        if (this.load && this.src) {
-          var img = new Image();
-          img.src = this.src;
-          this.loading = true;
-          img.onload = function() {
-            this._setSrc(this.src);
-            this.loading = false;
-            if (this.fade) {
-              this._placeholderEl.classList.add('fadein');
-            }
-          }.bind(this);
-        }
+    srcChanged: function(newSrc, oldSrc) {
+      if (newSrc !== oldSrc) {
+        this.loaded = false;
+      }
+    },
+
+    placeholderBackgroundUrlChanged: function() {
+      this.$.placeholder.style.backgroundImage =
+        this.placeholderBackgroundUrl;
+    },
+
+    transformChanged: function() {
+      var placeholderStyle = this.$.placeholder.style;
+
+      this.style.backgroundSize =
+        placeholderStyle.backgroundSize = this.sizing;
+
+      this.style.backgroundPosition =
+        placeholderStyle.backgroundPosition =
+        this.sizing ? this.position : null;
+
+      this.style.backgroundRepeat =
+        placeholderStyle.backgroundRepeat =
+        this.sizing ? 'no-repeat' : null;
+    },
+
+    loadBehaviorChanged: function() {
+      var img;
+      
+      if (!this.canLoad) {
+        return;
+      }
+
+      if (this.requiresPreload) {
+        img = new Image();
+        img.src = this.src;
+
+        this.loading = true;
+
+        img.onload = function() {
+          this.loading = false;
+          this.loaded = true;
+        }.bind(this);
       } else {
-        this._setSrc(this.src);
+        this.loaded = true;
       }
     },
 
-    _setSrc: function(src) {
+    loadStateChanged: function() {
+      if (this.requiresPreload) {
+        return;
+      }
+
       if (this.sizing) {
-        this.style.backgroundImage = src ? 'url(' + src + ')': '';
+        this.style.backgroundImage = this.src ? 'url(' + this.src + ')': '';
       } else {
-        this.$.img.src = src || '';
+        this.$.img.src = this.src || '';
```

Comment:
Template binding in this case would result in unwanted requests for the default `src` value.